policy/modules/kernel/devices.if

```diff
@@ -319,6 +319,24 @@ interface(`dev_create_generic_dirs',`
 	create_dirs_pattern($1, device_t, device_t)
 ')
 
+########################################
+## <summary>
+##	Watch generic device directories.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`dev_watch_generic_dirs',`
+	gen_require(`
+		type device_t;
+	')
+
+	allow $1 device_t:dir watch_dir_perms;
```

Comment:
Maybe this should rather be `watch_dirs_pattern($1, device_t, device_t)` to ensure that the domain has also search permissions on `device_t` (which I assume is needed to watch e.g. /dev subdirectories).

Reply:
This comment is possibly relevant in other interfaces, too.